pkg/sdk/testint/functions_integration_test.go

```diff
@@ -281,89 +1381,175 @@ func TestInt_OtherFunctions(t *testing.T) {
 		require.Equal(t, nid.Name(), e.Name)
 	})
 
-	t.Run("alter function: set log level", func(t *testing.T) {
-		f := createFunctionForSQLHandle(t, true, true)
-
+	t.Run("alter function: set and unset all for Java", func(t *testing.T) {
+		f, fCleanup := testClientHelper().Function.CreateJava(t)
+		t.Cleanup(fCleanup)
 		id := f.ID()
-		err := client.Functions.Alter(ctx, sdk.NewAlterFunctionRequest(id).WithSetLogLevel(string(sdk.LogLevelDebug)))
-		require.NoError(t, err)
-		assertFunction(t, id, false, true)
-	})
 
-	t.Run("alter function: unset log level", func(t *testing.T) {
-		f := createFunctionForSQLHandle(t, true, true)
+		assertions.AssertThatObject(t, objectassert.Function(t, id).
+			HasName(id.Name()).
+			HasDescription(sdk.DefaultFunctionComment),
+		)
 
-		id := f.ID()
-		err := client.Functions.Alter(ctx, sdk.NewAlterFunctionRequest(id).WithUnsetLogLevel(true))
-		require.NoError(t, err)
-		assertFunction(t, id, false, true)
-	})
+		assertions.AssertThatObject(t, objectassert.FunctionDetails(t, id).
+			HasExternalAccessIntegrationsNil().
+			HasSecretsNil(),
+		)
+
+		assertions.AssertThatObject(t, objectparametersassert.FunctionParameters(t, id).
+			HasAllDefaults().
+			HasAllDefaultsExplicit(),
+		)
 
-	t.Run("alter function: set trace level", func(t *testing.T) {
-		f := createFunctionForSQLHandle(t, true, true)
+		request := sdk.NewAlterFunctionRequest(id).WithSet(*sdk.NewFunctionSetRequest().
+			WithEnableConsoleOutput(true).
+			WithExternalAccessIntegrations([]sdk.AccountObjectIdentifier{externalAccessIntegration}).
+			WithSecretsList(*sdk.NewSecretsListRequest([]sdk.SecretReference{{VariableName: "abc", Name: secretId}})).
+			WithLogLevel(sdk.LogLevelWarn).
+			WithMetricLevel(sdk.MetricLevelAll).
+			WithTraceLevel(sdk.TraceLevelAlways).
+			WithComment("new comment"),
+		)
 
-		id := f.ID()
-		err := client.Functions.Alter(ctx, sdk.NewAlterFunctionRequest(id).WithSetTraceLevel(string(sdk.TraceLevelAlways)))
+		err := client.Functions.Alter(ctx, request)
 		require.NoError(t, err)
-		assertFunction(t, id, false, true)
-	})
 
-	t.Run("alter function: unset trace level", func(t *testing.T) {
-		f := createFunctionForSQLHandle(t, true, true)
+		assertions.AssertThatObject(t, objectassert.Function(t, id).
+			HasName(id.Name()).
+			HasDescription("new comment"),
+		)
 
-		id := f.ID()
-		err := client.Functions.Alter(ctx, sdk.NewAlterFunctionRequest(id).WithUnsetTraceLevel(true))
+		assertions.AssertThatObject(t, objectassert.FunctionDetails(t, id).
+			HasExternalAccessIntegrations(fmt.Sprintf(`[%s]`, externalAccessIntegration.FullyQualifiedName())).
```

Comment:
Can we have an extension function for these types and pass slices/maps there?

Reply:
Yes, I will add them